File: frontend/src/types/organization.ts
/**
 * Organization types.
 */

export interface BillingContactSummary {
  id: string;
  name: string;
  email: string;
  role_title: string | null;
}

export interface Organization {
  id: string;
  name: string;
  aliases: string[] | null;
  billing_contact_id: string | null;
  address_street: string | null;
  address_city: string | null;
  address_state: string | null;
  address_zip: string | null;
  address_country: string | null;
  inventory_url: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
  monday_id: string | null;
  monday_last_synced: string | null;
}

export interface OrganizationCreate {
  name: string;
  aliases?: string[];
  billing_contact_id?: string;
  address_street?: string;
  address_city?: string;
  address_state?: string;
  address_zip?: string;
  address_country?: string;
  inventory_url?: string;
  notes?: string;
}

export interface OrganizationUpdate {
  name?: string;
  aliases?: string[];
  billing_contact_id?: string | null;
  address_street?: string | null;
  address_city?: string | null;
  address_state?: string | null;
  address_zip?: string | null;
  address_country?: string | null;
  inventory_url?: string | null;
  notes?: string | null;
}

export interface ProjectSummaryForOrg {
  id: string;
  name: string;
  status: string;
  start_date: string;
  end_date: string | null;
}

export interface ContactSummaryForOrg {
  id: string;
  name: string;
  email: string;
  role_title: string | null;
}

export interface OrganizationDetail extends Organization {
  project_count: number;
  projects: ProjectSummaryForOrg[];
  contacts: ContactSummaryForOrg[];
  billing_contact: BillingContactSummary | null;
}

File: backend/app/schemas/organization.py
"""Organization Pydantic schemas."""

from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ProjectSummaryForOrg(BaseModel):
    """Minimal project info for organization detail."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    name: str
    status: str  # Use str to avoid circular import with ProjectStatus
    start_date: date
    end_date: date | None = None


class ContactSummaryForOrg(BaseModel):
    """Minimal contact info for organization detail."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    name: str
    email: str
    role_title: str | None = None


class OrganizationBase(BaseModel):
    """Base organization schema."""

    name: str = Field(..., min_length=1, max_length=255)
    aliases: list[str] | None = None


class OrganizationCreate(OrganizationBase):
    """Schema for creating an organization."""

    pass


class OrganizationUpdate(BaseModel):
    """Schema for updating an organization."""

    name: str | None = Field(None, min_length=1, max_length=255)
    aliases: list[str] | None = None


class OrganizationResponse(OrganizationBase):
    """Organization response schema."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    created_at: datetime
    updated_at: datetime


class OrganizationDetail(OrganizationResponse):
    """Detailed organization response with counts."""

    project_count: int = 0


class OrganizationDetailWithRelations(OrganizationDetail):
    """Organization with nested projects and contacts."""

    projects: list[ProjectSummaryForOrg] = []
    contacts: list[ContactSummaryForOrg] = []
